js/main.js

```diff
@@ -39,3 +39,124 @@ const getRandomDecimal = (min, max, precision) => {
+
+  `Sasazuka Station is 5 minutes by train to Shinjuku Station,
+  It is a very convenient location because you can use the metropolitan Shinjuku line besides the Keio line! !
+  The recommended neighboring places are Yoyogi Park and Shimokitazawa Station.
+  Yoyogi Park is a big park where you can feel the four seasons!
+  Shimokitazawa station is old clothes and music, since the night in the popularity of the`,
+];
+
+const OFFER_PHOTOS = [
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg',
+];
+
+/**
+ * Return a random element from an array
+ * @param {array} arr - array of any length;
+ * @returns {any} random element
+ */
+const getRandomElement = (arr) => arr[getRandomInt(0, arr.length - 1)];
+
+/**
+ * Return a random subarray from given array and length
+ * @param {array} arr - array of any length;
+ * @param {integer} - length of result array
+ * @returns {array} new subarray;
+ */
+const getRandomSubArray = (arr, arrLength) => {
+  const resultArr = [];
+  for (let int = 0; int < arrLength; int++) {
+    const randElem = getRandomElement(arr);
+
+    if (!resultArr.includes(randElem)) {
+      resultArr.push(randElem);
+    }
+  }
+  return resultArr;
+};
+
+/**
+ * Generates a random offer from given list of data
+ * @param {object} location - location object with latitude and longitude values;
+ * @returns {object} random offer
+ */
+const getRandOffer = (location) => {
+  const offer = {};
+  offer.title = getRandomElement(OFFER_TITLE);
+  offer.address = `${location.lat}, ${location.lng}`;
+  offer.price = getRandomInt(1500, 5000);
+  offer.type = getRandomElement(OFFER_TYPES);
+  offer.rooms = getRandomInt(1, 10);
+  offer.guests = getRandomInt(1, 10);
+  offer.checkin = getRandomElement(OFFER_CHECK_TIMES);
+  offer.checkout = getRandomElement(OFFER_CHECK_TIMES);
+  offer.features = getRandomSubArray(OFFER_FEATURES, getRandomInt(1, OFFER_FEATURES.length - 1));
+  offer.description = getRandomElement(OFFER_DESCRIPTIONS);
+  offer.photos = getRandomSubArray(OFFER_PHOTOS, getRandomInt(2, 4));
+  return offer;
+};
+
+/**
+ * Generates a random advertisement from given list of data
+ * @returns {object} random advert
+ */
+const getRandomAdvert = () => {
+  const advert = {};
+
+  const randomAuthor = getRandomElement(AUTHORS);
+  advert.author = randomAuthor;
+
+  const randLocation = {};
+  randLocation.lat = getRandomDecimal(35.65, 35.7, 5);
+  randLocation.lng = getRandomDecimal(139.7, 139.8, 5);
+  advert.location = randLocation;
+
+  advert.offer = getRandOffer(randLocation);
+
+  return advert;
+};
+
+const RANDOM_ADVERT_COUNT = 10;
+
+const randomAdverts = new Array(RANDOM_ADVERT_COUNT).fill(null).map(() => getRandomAdvert());
```

Comment:
```suggestion
const randomAdverts = new Array(RANDOM_ADVERT_COUNT).fill(null).map(getRandomAdvert);
```

зачем создавать новую функцию, в которой ты вызываешь другую функцию? передай просто ссылку на существующую функцию

Reply:
не исправлено. Хотя я тут даже пример кода указал.

Reply:
исправил!